File: Programmers/Lv2_탑.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;

vector<int> solution(vector<int> heights) {
	vector<int> answer;
	stack<int> s;
	int size = heights.size();
	int index = heights.size();
	int j = 0;

	for (int i = size - 1; i >= 1; i--) {
		int temp = heights[i];
		for (j = i - 1; j >= 0; j--) {
			if (temp < heights[j]) {
				s.push(j + 1);
				break;
			}
		}
		if (j == -1)
			s.push(0);
	}
	s.push(0);
	while (!s.empty()) {
		answer.push_back(s.top());
		s.pop();
	}

	return answer;
}

File: Programmers/Lv2/Lv2_탑.py
def solution(heights):
    answer = []
    for i in range(len(heights)-1, 0, -1):
        flag = True
        for j in range(i-1, -1, -1):
            if heights[j]>heights[i]:
                answer.append(j+1)
                flag = False
                break;
        if flag:
            answer.append(0)
    answer.append(0)
    answer.reverse()
    return answer

print(solution(	[6, 9, 5, 7, 4]))